Tests/JetworkingTests/ClientTests/ExecuterTests/AsyncRequestExecuterTests.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  File.swift
+//  
+//
+//  Created by Jens on 28.10.20.
+//
+
+import Foundation
+@testable import Jetworking
+import XCTest
+
+class AsyncRequestExecuterTests: XCTestCase {
+    var session: URLSession {
+        let configuration = URLSessionConfiguration.default
+        configuration.protocolClasses = [MockURLProtocol.self]
+        return URLSession.init(configuration: configuration)
+    }
+
+    func testOrderDueToAsyncRequestExecuter() {
+        let requestExecuter: AsyncRequestExecutor = .init(session: session)
+
+        MockURLProtocol.requestHandler = { request in
+            let waitTime: [String: TimeInterval] = [
+                "https://www.jamitlabs.com/somePath0": 2,
+                "https://www.jamitlabs.com/somePath1": 0,
+                "https://www.jamitlabs.com/somePath2": 3,
+                "https://www.jamitlabs.com/somePath3": 1
+            ]
+
+            guard
+                let url = request.url,
+                let timeToWait = waitTime[url.absoluteString]
+            else {
+                throw MockURLProtocol.MockError.invalidConfiguration
+            }
+
+            let response: HTTPURLResponse = HTTPURLResponse(
+                url: url,
+                statusCode: 200,
+                httpVersion: nil,
+                headerFields: nil
+            )!
+
+            return (response, nil, timeToWait)
+        }
+
+        let firstExpectation = expectation(description: "Wait for first get")
+        _ = requestExecuter.send(
+            request: URLRequest(
+                url: URL(string: "https://www.jamitlabs.com/somePath0")!,
```

Comment:
Why don't you access the `waitTime` dictionary here but rather use the string again?

Reply:
I've extracted the paths into separate variables to use instead of the strings. Hopefully this solution is ok. Creating an array could be another solution but would mess up the code with index access.

Reply:
Okay for me.